### PR TITLE
[Draft] Distinct

### PR DESCRIPTION
**Pull request by nathansoz**

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Security.Cryptography.X509Certificates;
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs]: >System.Security.Cryptography.X509Certificates; [](http://example.com/codeflow?start=6&length=46)

curious what we need this for?

File: src/libraries/Microsoft.PowerFx.Core/Texl/BuiltinFunctionsCore.cs [hunk 2]
```diff
@@ -1,219 +1,220 @@
...
+        public static readonly TexlFunction DateDiff = _library.Append(new DateDiffFunction());
+        public static readonly TexlFunction DateDiffT = _library.Append(new DateDiffTFunction());
+        public static readonly TexlFunction DateTimeValue = _library.Append(new DateTimeValueFunction());
+        public static readonly TexlFunction DateTimeValue_UO = _library.Append(new DateTimeValueFunction_UO());
+        public static readonly TexlFunction DateValue = _library.Append(new DateValueFunction());
+        public static readonly TexlFunction DateValue_UO = _library.Append(new DateValueFunction_UO());
+        public static readonly TexlFunction Day = _library.Append(new DayFunction());
+        public static readonly TexlFunction Degrees = _library.Append(new DegreesFunction());
+        public static readonly TexlFunction DegreesT = _library.Append(new DegreesTableFunction());
+        public static readonly TexlFunction Distinct = _library.Append(new DistinctFunction());
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Texl/BuiltinFunctionsCore.cs]: >_library [](http://example.com/codeflow?start=55&length=8)

We may need to register this differently to ensure that it doesn't conflict with PowerApps' definition.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs (hunk 1 above)
Comment (nathansoz) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs]: I think it got pulled in by VS with a incorrect autocomplete suggestion. I'll remove.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Distinct.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+>> CountRows(Distinct([{a: 1}, {a: 1}, {a: 3}], Value.a))
+2
+
+>> Distinct([{a: 1}, {a: 1}, {a: 3}], Value.a)
+Table({Result:1},{Result:3})
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Distinct.txt]: >Result [](http://example.com/codeflow?start=7&length=6)

fyi - we'll trying to standardize on using "Value"  name, so we may end up changing this to be Value instead of Result.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Distinct.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+>> CountRows(Distinct([{a: 1}, {a: 1}, {a: 3}], Value.a))
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Distinct.txt]: .txt file is great. 
We'll eventually need to add more cases here,  including corner cases like blank/error.

File: src/libraries/Microsoft.PowerFx.Core/Texl/BuiltinFunctionsCore.cs (hunk 2 above)
Comment (nathansoz) [reply src/libraries/Microsoft.PowerFx.Core/Texl/BuiltinFunctionsCore.cs]: Will wait for your advice in the new PR.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Distinct.txt (hunk 3 above)
Comment (nathansoz) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Distinct.txt]: Talked offline. Leaving value for now, easy to change later.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Distinct.txt (hunk 4 above)
Comment (nathansoz) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Distinct.txt]: Added a bunch of cases. :)
